SimG4CMS/Forward/src/BHMNumberingScheme.cc

```diff
@@ -6,43 +6,20 @@
 
 BHMNumberingScheme::BHMNumberingScheme() { LogDebug("BHMSim") << " Creating BHMNumberingScheme"; }
 
-int BHMNumberingScheme::detectorLevel(const G4Step* aStep) const {
-  //Find number of levels
-  const G4VTouchable* touch = aStep->GetPreStepPoint()->GetTouchable();
-  return (touch) ? ((touch->GetHistoryDepth()) + 1) : 0;
-}
-
-void BHMNumberingScheme::detectorLevel(const G4Step* aStep, int& level, int* copyno, G4String* name) const {
-  //Get name and copy numbers
-  if (level > 0) {
-    const G4VTouchable* touch = aStep->GetPreStepPoint()->GetTouchable();
-    for (int ii = 0; ii < level; ++ii) {
-      int i = level - ii - 1;
-      name[ii] = ForwardName::getName(touch->GetVolume(i)->GetName());
-      copyno[ii] = touch->GetReplicaNumber(i);
-    }
-  }
-}
-
 unsigned int BHMNumberingScheme::getUnitID(const G4Step* aStep) const {
   unsigned intindex = 0;
-  int level = detectorLevel(aStep);
 
-  LogDebug("BHMSim") << "BHMNumberingScheme number of levels= " << level;
-  if (level > 0) {
-    int* copyno = new int[level];
-    G4String* name = new G4String[level];
-    detectorLevel(aStep, level, copyno, name);
+  //Find number of levels
+  const G4VTouchable* touch = aStep->GetPreStepPoint()->GetTouchable();
+  int level = (touch) ? ((touch->GetHistoryDepth()) + 1) : 0;
 
-    if (level > 3) {
-      int subdet = copyno[0];
-      int zside = copyno[3];
-      int station = copyno[1];
-      intindex = packIndex(subdet, zside, station);
-      LogDebug("BHMSim") << "BHMNumberingScheme : subdet " << subdet << " zside " << zside << " station " << station;
-    }
-    delete[] copyno;
-    delete[] name;
+  LogDebug("BHMSim") << "BHMNumberingScheme number of levels= " << level;
+  if (level > 3) {
```

Comment:
In previous implementation if(level>3) the fix depth is used not depended on level of 'level'

Reply:
Not sure what you mean. In [previous implementation](https://github.com/cms-sw/cmssw/blob/master/SimG4CMS/Forward/src/BHMNumberingScheme.cc#L32-L46) if level was less than or equal to 3, the arrays filled by `::detectorLevel` were not used - unless the calls in that function had side-effects